inst/include/cpp11/r_vector.hpp

```diff
@@ -572,9 +575,9 @@ inline typename cpp11::r_vector<T>::const_iterator cpp11::r_vector<T>::find(
 template <typename T>
 inline T r_vector<T>::const_iterator::operator*() const {
   if (data_->is_altrep()) {
-    return buf_[pos_ - block_start_];
+    return static_cast<T>(buf_[pos_ - block_start_]);
   } else {
-    return data_->data_p_[pos_];
+    return static_cast<T>(data_->data_p_[pos_]);
```

Comment:
We probably don't need these, the implicit casts to `T` will kick in as needed (for `T = r_bool` and `T = uint8_t`) but it feels like good practice to be explicit?